iothub_client/src/iothub_client_core.c

```diff
@@ -813,7 +817,7 @@ static int ScheduleWork_Thread(void* threadArgument)
             /*Codes_SRS_IOTHUBCLIENT_01_040: [If acquiring the lock fails, IoTHubClientCore_LL_DoWork shall not be called.]*/
             /*no code, shall retry*/
         }
-        (void)ThreadAPI_Sleep(1);
+        (void)ThreadAPI_Sleep(iotHubClientInstance->loop_timeout);
```

Comment:
again, loop_timeout is quite different from what we usually call it. Should be something that matches the other names  commented above.

Reply:
addressed and removed race condition